mm2src/rpc/get_public_key_hash.rs

```diff
@@ -0,0 +1,36 @@
+use common::{mm_ctx::MmArc, mm_error::MmError, HttpStatusCode};
+use derive_more::Display;
+use http::StatusCode;
+use serde_json::Value as Json;
+
+#[derive(Serialize)]
+pub struct GetPublicKeyHashResponse {
+    public_key_hash: String,
+}
+#[derive(Serialize, Display, SerializeErrorType)]
+#[serde(tag = "error_type", content = "error_data")]
+pub enum GetPublicKeyHashError {
+    Internal(String),
+}
+
+pub type GetPublicKeyHashRpcResult<T> = Result<T, MmError<GetPublicKeyHashError>>;
+
+impl HttpStatusCode for GetPublicKeyHashError {
+    fn status_code(&self) -> StatusCode {
+        match self {
+            GetPublicKeyHashError::Internal(_) => StatusCode::INTERNAL_SERVER_ERROR,
+        }
+    }
+}
+
+pub async fn get_public_key_hash(ctx: MmArc, _req: Json) -> GetPublicKeyHashRpcResult<GetPublicKeyHashResponse> {
+    let public_key_hash = match ctx.secp256k1_key_pair.as_option() {
```

Comment:
First of all, please consider using [MmCtx::rmd160](https://github.com/KomodoPlatform/atomicDEX-API/blob/dev/mm2src/common/mm_ctx.rs#L153). It returns `RIPEMD160(SHA256(x)) where x is secp256k1 pubkey derived from passphrase`.
Secondly, if you want to return an error on `None`, you can use [OrMmError::or_mm_err](https://github.com/KomodoPlatform/atomicDEX-API/blob/dev/mm2src/common/mm_error/or_mm_error.rs#L20):
```rust
ctx.secp256k1_key_pair.as_option().or_mm_err(|| GetPublicKeyHashError::Internal("A valid public key is needed to get hashed version. Please provide one".to_string()))?;
```